spec/broadside/deploy/ecs_deploy_spec.rb

```diff
@@ -68,11 +68,59 @@
       expect { deploy.bootstrap }.to raise_error(/Service doesn't exist and no :service_config/)
     end
 
-    it 'succeeds' do
-      deploy.deploy_config.service_config = service_config
-      deploy.deploy_config.task_definition_config = task_definition_config
+    context 'with an existing task definition and service' do
+      let(:arn) { 'arn:aws:ecs:us-east-1:1234' }
+      let(:existing_service) do
+        {
+          service_name: task_name,
+          service_arn: "#{arn}:service/#{task_name}",
+          deployments: [{ desired_count: 1, running_count: 1 }]
+        }
+      end
+      let(:stub_service_response) { { services: [existing_service], failures: [] } }
+      let(:task_definition_arn) { "#{arn}:task-definition/#{task_name}:1" }
+      let(:stub_task_definition_response) { { task_definition_arns: [task_definition_arn] } }
+      let(:stub_describe_task_definition_response) do
+        {
+          task_definition: {
+            task_definition_arn: task_definition_arn,
+            container_definitions: [
+              {
+                name: family
+              }
+            ],
+            family: family
+          }
+        }
+      end
+
+      before(:each) do
+        ecs_stub.stub_responses(:list_task_definitions, stub_task_definition_response)
+        ecs_stub.stub_responses(:describe_task_definition, stub_describe_task_definition_response)
+        ecs_stub.stub_responses(:describe_services, stub_service_response)
+      end
+
+      it 'succeeds' do
+        deploy.deploy_config.service_config = service_config
+        deploy.deploy_config.task_definition_config = task_definition_config
 
-      expect { deploy.bootstrap }.to_not raise_error
+        expect { deploy.bootstrap }.to_not raise_error
+      end
+
+      context 'and some configured bootstrap commands' do
+        before do
+          Broadside.configure do |config|
+            config.deploy.targets[task_name.to_sym][:bootstrap_commands] = [
+              %w(foo bar baz)
+            ]
+          end
+        end
+
+        it 'runs bootstrap commands' do
+          expect(deploy).to receive(:run_command).with(%w(foo bar baz))
```

Comment:
this is a sufficient spec, but just in case you didn't notice, you can actually go access every single parameter to every single AWS call in the `api_request_log` variable: https://github.com/lumoslabs/broadside/blob/4abc5e65f0205b03826a7a461f803125a2038007/spec/broadside/deploy/ecs_deploy_spec.rb#L150

i don't know if you need that really - you probably don't - though there are no specs for a lot of functionality (there were no specs at all when i started doing the work on broadside)